File: LeetCode/Easy/Widest Vertical Area Between Two Points Containing No Points/1637.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var maxWidthOfVerticalArea = function(points) {
    const n = points.length;
    const xArr = new Array(n);
    for (let i = 0; i < n; i++) {
        xArr[i] = points[i][0];
    }

    xArr.sort((a, b) => a - b);

    let maxWidth = 0;
    for (let i = 1; i < n; i++) {
        const width = xArr[i] - xArr[i - 1];
        if (width > maxWidth) {
            maxWidth = width;
        }
    }

    return maxWidth;
};

File: LeetCode/Easy/Widest Vertical Area Between Two Points Containing No Points/1637.java
class Solution {
    public int maxWidthOfVerticalArea(int[][] points) {
        int n = points.length;
        int[] xArr = new int[n];
        for (int i = 0; i < n; i++) {
            xArr[i] = points[i][0];
        }

        Arrays.sort(xArr);

        int maxWidth = 0;
        for (int i = 1; i < n; i++) {
            int width = xArr[i] - xArr[i - 1];
            if (width > maxWidth) {
                maxWidth = width;
            }
        }

        return maxWidth;
    }
}
